From 27aeb4f35f0599398043434905f149249edde832 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:34:58 -0400
Subject: [alsa] Revert: hda: mask buggy stream DMA0 for Broadwell display controller

Message-id: <140972823573-333-patchset-jkysela@redhat.com>
Patchwork-id: 91029
O-Subject: [RHEL 7.1 PATCH 332/442] ALSA: Revert "ALSA: hda - mask buggy stream DMA0 for Broadwell display controller"
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

This reverts commit 7189eb9b8f7962474956196c301676470542f253.

It will use LPIB to get the DMA position on Broadwell HDMI Audio.

Signed-off-by: Libin Yang <libin.yang@intel.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: a49d4d7c6edaaaaed92c8764ac18a0b58ffa09c7)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_intel.c b/sound/pci/hda/hda_intel.c
index 920e314..0a6b561 100644
--- a/sound/pci/hda/hda_intel.c
+++ b/sound/pci/hda/hda_intel.c
@@ -1373,12 +1373,6 @@ static int azx_first_init(struct azx *chip)
  /* initialize streams */
  azx_init_stream(chip);
 
- /* workaround for Broadwell HDMI: the first stream is broken,
-  * so mask it by keeping it as if opened
-  */
- if (pci->vendor == 0x8086 && pci->device == 0x160c)
-  chip->azx_dev[0].opened = 1;
-
  /* initialize chip */
  azx_init_pci(chip);
  azx_init_chip(chip, (probe_only[dev] & 2) == 0);
-- 
1.7.1